client/app/services/parameters/Parameter.js

```diff
@@ -0,0 +1,146 @@
+import { isNull, isObject, isFunction, isUndefined, isEqual, has } from 'lodash';
+import {
+  TextParameter, NumberParameter, EnumParameter, QueryBasedDropdownParameter,
+  DateParameter, DateRangeParameter,
+} from '.';
+
+class Parameter {
+  constructor(parameter, parentQueryId) {
+    this.title = parameter.title;
+    this.name = parameter.name;
+    this.type = parameter.type;
+    this.global = parameter.global; // backward compatibility in Widget service
+    this.parentQueryId = parentQueryId;
+
+    // Used for meta-parameters (i.e. dashboard-level params)
+    this.locals = [];
+
+    // Used for URL serialization
+    Object.defineProperty(this, 'urlPrefix', {
```

Comment:
As Cypress pointed out, this doesn't work with inheritance :slightly_smiling_face: